mmv1/products/cloudbuild/api.yaml

```diff
@@ -25,6 +25,138 @@ apis_required:
+          If you need to change it, please create another BitbucketServerConfig.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'secrets'
+        required: true
+        description: |
+          Secret Manager secrets needed by the config.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'adminAccessTokenVersionName'
+            required: true
+            description: |
+              The resource name for the admin access token's secret version.
+          - !ruby/object:Api::Type::String
+            name: 'readAccessTokenVersionName'
+            required: true
+            description: |
+              The resource name for the read access token's secret version.
+          - !ruby/object:Api::Type::String
+            name: 'webhookSecretVersionName'
+            required: true
+            input: true
+            description: |
+              Immutable. The resource name for the webhook secret's secret version. Once this field has been set, it cannot be changed.
+              If you need to change it, please create another BitbucketServerConfig.
+      - !ruby/object:Api::Type::Time
+        name: 'createTime'
+        output: true
+        description: |
+          Time when the config was created.
+      - !ruby/object:Api::Type::String
+        name: 'username'
+        required: true
+        description: |
+          Username of the account Cloud Build will use on Bitbucket Server.
+      - !ruby/object:Api::Type::String
+        name: 'webhookKey'
+        output: true
+        description: |
+          Output only. UUID included in webhook requests. The UUID is used to look up the corresponding config.
+      - !ruby/object:Api::Type::String
+        name: 'apiKey'
+        required: true
+        input: true
+        description: |
+          Immutable. API Key that will be attached to webhook. Once this field has been set, it cannot be changed.
+          If you need to change it, please create another BitbucketServerConfig.
+      - !ruby/object:Api::Type::Array
+        name: 'connectedRepositories'
+        description: |
+          Connected Bitbucket Server repositories for this config.
+        item_type: !ruby/object:Api::Type::NestedObject
+          name: 'BitbucketServerRepositoryId'
+          description: |
+            BitbucketServerRepositoryId identifies a specific repository hosted on a Bitbucket Server.
+          properties:
+            - !ruby/object:Api::Type::String
+              name: 'projectKey'
+              required: true
+              description: |
+                Identifier for the project storing the repository.
+            - !ruby/object:Api::Type::String
+              name: 'repoSlug'
+              required: true
+              description: |
+                Identifier for the repository.
+      - !ruby/object:Api::Type::String
+        name: 'peeredNetwork'
```

Comment:
Would it be possible to add a test that uses the `peered_network` and `ssl_ca` fields?

Reply:
I attempted this, and ended up only adding an extra example while skipping test generation. 

tl;dr -  I would need to use `google_compute_network`, and it seems to be unable to delete itself.

You can see in the peered_network example I added that I need to use the compute api and service networking apis to create dependant resources. When testing manually, things work until `terraform destroy`. It seems that while using the `google_compute_network`, firewall rules are created for that network, which then causes issues on deletion. The test fails to clean up the resources.

Can we move on with just the extra docs?

Reply:
Could you try using `BootstrapSharedTestNetwork`? There is an example [here](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/vertexai/terraform.yaml#L58).